# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gnu-efi/elf-Ia32_efi_lds.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -aur gnu-efi-3.0/gnuefi/elf_ia32_efi.lds gnu-efi-3.0-fix/gnuefi/elf_ia32_efi.lds
--- gnu-efi-3.0/gnuefi/elf_ia32_efi.lds	2008-09-18 12:52:58.000000000 -0400
+++ gnu-efi-3.0-fix/gnuefi/elf_ia32_efi.lds	2009-07-16 19:15:14.658001000 -0400
@@ -13,9 +13,15 @@
    *(.text.*)
    *(.gnu.linkonce.t.*)
   }
-  .reloc :
+  . = ALIGN(4096);
+  .sdata :
   {
-   *(.reloc)
+   *(.got.plt)
+   *(.got)
+   *(.srodata)
+   *(.sdata)
+   *(.sbss)
+   *(.scommon)
   }
   . = ALIGN(4096);
   .data :
@@ -50,6 +56,11 @@
     *(.data.rel*)
   }
   . = ALIGN(4096);
+  .reloc :		/* This is the PECOFF .reloc section! */
+  {
+    *(.reloc)
+  }
+  . = ALIGN(4096);
   .dynsym   : { *(.dynsym) }
   . = ALIGN(4096);
   .dynstr   : { *(.dynstr) }
@@ -58,5 +69,8 @@
   {
     *(.rel.reloc)
     *(.eh_frame)
+    *(.note.GNU-stack)
   }
+  .comment 0 : { *(.comment) }
 }
+

